hs/CalendarResponse.hs

```diff
@@ -0,0 +1,281 @@
+                                      (T.pack sect)
+                                      (T.pack session)
+    else getData $ returnTutorialTimes (T.pack code)
+                                       (T.pack sect)
+                                       (T.pack session)
+    where
+        getData info = do
+            courseInfo <- info 
+            return (courseInfo, code, session, sect)
+
+-- ** Event Creation
+
+-- | Generates an event for each course.
+getEvent :: String -> ([Time], String, String, String) -> [String]
+getEvent timeSystem (timeFields, code, session, sect) = 
+    concat $ eventsByCourse code session sect start end dates timeSystem
+    where
+        dataInOrder = orderData timeFields
+        start = startTime dataInOrder
+        end = endTime dataInOrder
+        dates = getDatesByCourse session dataInOrder
+
+-- | Creates an event for each course depending on its session.
+eventsByCourse :: String -- ^ Course code.
+            -> String -- ^ Course session.
+            -> String -- ^ Course section.
+            -> [[String]] -- ^ Start times.
+            -> [[String]] -- ^ End times.
+            -> [[(String, String)]] -- ^ Start/End dates.
+            -> String -- ^ Current time.
+            -> [[String]]
+eventsByCourse code session sect start end dates timeSystem =
+    if session == "Y" then year else halfCourse
+    where
+        year = halfFall ++ halfWinter
+        halfFall = eventsByTime code sect start end (dates !! 0) timeSystem
+        halfWinter = eventsByTime code sect start end (dates !! 1) timeSystem
+        halfCourse = eventsByTime code sect start end (dates !! 0) timeSystem
+
+-- | Creates an event for each start/end time and date.
+eventsByTime :: String -- ^ Course code.
+             -> String  -- ^ Course section.
+             -> [[String]] -- ^ Start times.
+             -> [[String]] -- ^ End times.
+             -> [(String, String)] -- ^ Start/End dates.
+             -> String -- ^ Current time.
+             -> [[String]]
+eventsByTime code sect start end dates timeSystem =
+    map (eventsByDate code sect timeSystem) (zip' start end dates)
+
+-- | Join three lists together in tuples of three elements
+zip' :: [a] -> [b] -> [c] -> [(a,b,c)]
+zip' (x:xs) (y2:ys) (z:zs) = (x,y2,z):zip' xs ys zs
+zip' _ _ _ = []
+
+-- | Creates an event for each start/end time.
+eventsByDate :: String -- ^ Course code.
+             -> String -- ^ Course section.
+             -> String -- ^ Current time.
+             -> ([String], -- ^ Start time
+             [String], -- ^ End time
+             (String, String)) -- ^ Start/End date.
+             -> [String]
+eventsByDate code sect timeSystem (start, end, dates) =
+    (zip start end) >>= (eventsGenerator code sect timeSystem dates)
+
+-- | Generates the string that represents each event.
+eventsGenerator :: String -- ^ Course code.
+                -> String -- ^ Course section.
+                -> String -- ^ Current time.
```

Comment:
I called the variable timeSystem because the value to which it refers comes from calling the following function.
getCurrentTime :: IO UTCTimeSource
Get the current UTC time from the system clock.